src/fides/api/ops/models/privacy_preference.py

```diff
@@ -69,18 +93,24 @@ class PrivacyPreferenceHistory(Base):
             padding="pkcs5",
         ),
     )
+    # Whether the user wants to opt in, opt out, or has acknowledged the notice
     preference = Column(EnumColumn(UserConsentPreference), nullable=False, index=True)
+    # The specific historical record the user consented to
     privacy_notice_history_id = Column(
         String, ForeignKey(PrivacyNoticeHistory.id), nullable=False, index=True
     )
+    # The privacy request created to propage the preferences
     privacy_request_id = Column(
         String, ForeignKey(PrivacyRequest.id, ondelete="SET NULL"), index=True
     )
+    # Optional FK to a verified provided identity (like email or phone), if applicable
     provided_identity_id = Column(String, ForeignKey(ProvidedIdentity.id), index=True)
-    relevant_systems = Column(
-        MutableList.as_mutable(ARRAY(String))
-    )  # Systems whose data use match.  This doesn't necessarily mean we propagate. Some may be intentionally skipped later.
+
+    # Systems whose data use match.  This doesn't necessarily mean we propagate. Some may be intentionally skipped later.
+    relevant_systems = Column(MutableList.as_mutable(ARRAY(String)))
+    # Location where we received the request
     request_origin = Column(EnumColumn(RequestOrigin))
```

Comment:
> one question I have about this RequestOrigin enum- overlay corresponds to banner or modal components?

The way I understand it, both.  The Overlay is where the user makes their consent choices.  The Privacy Experience determines whether that overlay was triggered by a banner or a link.  Separately, when I get the Privacy Experiences backend, we can start tracking here specifically which privacy experience the user saw too.

> And in which scenarios (from a customer-facing perspective) will api be used?

I don't think it will be used for anything yet, I've been asked to say that none of our privacy notice templates are "api".
https://github.com/ethyca/fides/pull/3120#discussion_r1175586491. From Michael: " API flag here is to support some future validation logic that allows a customer to create a notice for presentation in a different medium instead of the privacy center or overlay."

> Also, particularly with GPC (and other scenarios too?), we'll automatically send consent preference downstream to fides api. In this case, what does the request origin map to?

RequestOrigin is nullable right now, so in this case, I'd put nothing.  If we need to add more fields to this RequestOrigin enum to support these use cases we can do that.